### PR TITLE
Jc/add view mentor placements

### PR DESCRIPTION
**Pull request by JamieCleare2525**
## Context

Add show pages for mentors.

## Changes proposed in this pull request

_Placements_
- Views/Controllers/Routes/etc. for Support Users to view School Mentors
- Views/Controllers/Routes/etc. for School Users to view Mentors

## Guidance to review
_Placements_
- Sign in as Anne
- visit http://placements.localhost:3000/schools/:school_id/mentors/:id
- Check mentors `first_name`, `last_name`, `trn` are displayed

- Sign in as Colin
- visit http://placements.localhost:3000/support/schools/:school_id/mentors/:id
- Check mentors `first_name`, `last_name`, `trn` are displayed

## Link to Trello card

https://trello.com/c/z6xJxOoz/196-as-a-support-user-i-want-to-view-the-details-of-a-mentor
https://trello.com/c/oPOYv499/215-as-a-user-i-want-to-view-the-details-of-a-mentor

## Screenshots
_School User (Anne)_
Screenshot 2024-02-16 at 13 45 30

_Support User (Colin)_
Screenshot 2024-02-16 at 14 19 54

### CODE REVIEW

File: app/views/placements/schools/mentors/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+<div class="govuk-width-container">
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+      <span class="govuk-caption-l"><%= @school.name %></span>
+      <h2 class="govuk-heading-l"><%= @mentor.full_name %></h2>
+
+      <%= govuk_summary_list do |summary_list| %>
+        <% summary_list.with_row do |row| %>
+          <% row.with_key(text: t(".attributes.mentors.first_name")) %>
```
Comment (elceebee) [inline app/views/placements/schools/mentors/show.html.erb:17]: for first and last names, you might want to use `Mentor.human_attribute_name.first_name` (or last_name)

I think we will probably render the trn field in multiple ways so good to use a defined translation for that one (so we don't end up with the little problem we have with user email at the moment. Someone has defined the human attributes as "Email address", but sometimes we just want "Email")

**Comment by JamieCleare2525** [comment]
[As a Support User, I want to view the details of a Mentor](https://trello.com/c/z6xJxOoz/196-as-a-support-user-i-want-to-view-the-details-of-a-mentor)

**Comment by JamieCleare2525** [comment]
[As a User, I want to view the details of a Mentor](https://trello.com/c/oPOYv499/215-as-a-user-i-want-to-view-the-details-of-a-mentor)

**Review by CatalinVoineag — APPROVED** [review]

File: app/views/placements/support/schools/mentors/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+<%= render "placements/support/primary_navigation", current_navigation: :organisations %>
+
+<%= content_for(:before_content) do %>
+  <%= govuk_back_link(href: placements_support_school_mentors_path) %>
+<% end %>
+
+<div class="govuk-width-container">
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+      <span class="govuk-caption-l"><%= @school.name %></span>
```
Comment (elceebee) [inline app/views/placements/support/schools/mentors/show.html.erb:12]: For the support user, this is slightly different than for the normal user. It should be 
Mentors - school_name
image

**Review by elceebee — COMMENTED** [review]
Looking very nice! Just a couple of things

File: app/models/user_membership.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # == Schema Information
 #
-# Table name: memberships
+# Table name: user_memberships
```
[the remark this answers is not in the document]
Comment (JamieCleare2525) [reply app/models/user_membership.rb:3]: Annotate wasn't run on https://github.com/DFE-Digital/itt-mentor-services/pull/273

[the remark this answers is not in the document]
Comment (elceebee) [reply app/models/user_membership.rb:3]: Ah, I see.

File: app/views/placements/support/schools/mentors/show.html.erb (hunk 2 above)
Comment (JamieCleare2525) [reply app/views/placements/support/schools/mentors/show.html.erb]: resolved in [f66b7fb](https://github.com/DFE-Digital/itt-mentor-services/pull/274/commits/f66b7fbdbaca1bed6ce46733c6c8777cdf3e27aa)

File: app/views/placements/schools/mentors/show.html.erb (hunk 1 above)
Comment (JamieCleare2525) [reply app/views/placements/schools/mentors/show.html.erb]: resolved in [f66b7fb](https://github.com/DFE-Digital/itt-mentor-services/pull/274/commits/f66b7fbdbaca1bed6ce46733c6c8777cdf3e27aa)

File: config/locales/en/placements/schools/mentors.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+en:
+  placements:
+    schools:
+      mentors:
+        show:
+          attributes:
+            mentors:
+              first_name: First name
+              last_name: Last name
```
Comment (elceebee) [inline config/locales/en/placements/schools/mentors.yml:9]: So I think you can delete these two lines.

File: config/locales/en/placements/support/schools/mentors.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  placements:
+    support:
+      schools:
+        mentors:
+          show:
+            caption: "Mentors - %{school_name}"
+            attributes:
+              mentors:
+                first_name: First name
+                last_name: Last name
```
Comment (elceebee) [inline config/locales/en/placements/support/schools/mentors.yml:11]: And these two

**Review by elceebee — APPROVED** [review]
Good work! it would be good to delete the two unnecessary translations now that you are using human_attribute_name on those two fields. But I am very happy to approve as it is!

File: config/locales/en/placements/support/schools/mentors.yml (hunk 5 above)
Comment (JamieCleare2525) [reply config/locales/en/placements/support/schools/mentors.yml]: resolved in [3852d92](https://github.com/DFE-Digital/itt-mentor-services/pull/274/commits/3852d927370d77ecb3dd252d00e4ec3e140b9852)

File: config/locales/en/placements/schools/mentors.yml (hunk 4 above)
Comment (JamieCleare2525) [reply config/locales/en/placements/schools/mentors.yml]: resolved in [3852d92](https://github.com/DFE-Digital/itt-mentor-services/pull/274/commits/3852d927370d77ecb3dd252d00e4ec3e140b9852)

**Pull request merged**